File: features/fixtures/shared/utils/Logging.m
//
//  Logging.m
//  iOSTestApp
//
//  Created by Karl Stenerud on 02.11.23.
//  Copyright © 2023 Bugsnag. All rights reserved.
//

#import "Logging.h"

#if TARGET_OS_IPHONE
extern void i_kslog_logCBasic(const char *fmt, ...) __printflike(1, 2);
#endif

void logInternal(const char* level, NSString *format, va_list args) {
    NSString *formatted = [[NSString alloc] initWithFormat:format arguments:args];
    NSString *fullMessage = [NSString stringWithFormat:@"bugsnagci %s: %@", level, formatted];

    NSLog(@"%@", fullMessage);
    // TODO: Restore logging for macOS
#if TARGET_OS_IPHONE
    i_kslog_logCBasic("%s",
                          [[NSString stringWithFormat:@"%@ %@",
                            [NSDate date], fullMessage]
                           cStringUsingEncoding:NSUTF8StringEncoding]);
#endif
}

void logDebugObjC(NSString *format, ...) {
    va_list args;
    va_start(args, format);
    logInternal("debug", format, args);
    va_end(args);
}

void logInfoObjC(NSString *format, ...) {
    va_list args;
    va_start(args, format);
    logInternal("info", format, args);
    va_end(args);
}

void logWarnObjC(NSString *format, ...) {
    va_list args;
    va_start(args, format);
    logInternal("warn", format, args);
    va_end(args);
}

void logErrorObjC(NSString *format, ...) {
    va_list args;
    va_start(args, format);
    logInternal("error", format, args);
    va_end(args);
}

File: features/fixtures/shared/utils/Logging.swift
//
//  logging.swift
//  iOSTestApp
//
//  Created by Karl Stenerud on 02.11.23.
//  Copyright © 2023 Bugsnag. All rights reserved.
//

import Foundation

//public func logDebug(_ format: String) {
//    logDebug(format: format, args: 0 as Int)
//}
public func logDebug(_ format: String, _ args: CVarArg...) {
    withVaList(args) {
        logInternal("debug", format, $0)
    }
}

//public func logInfo(_ format: String) {
//    logInfo(format: format, args: 0 as Int)
//}
public func logInfo(_ format: String, _ args: CVarArg...) {
    withVaList(args) {
        logInternal("info", format, $0)
    }
}

//public func logWarn(_ format: String) {
//    logWarn(format: format, args: 0 as Int)
//}
public func logWarn(_ format: String, _ args: CVarArg...) {
    withVaList(args) {
        logInternal("warn", format, $0)
    }
}

//public func logError(_ format: String) {
//    logError(format: format, args: 0 as Int)
//}
public func logError(_ format: String, _ args: CVarArg...) {
    withVaList(args) {
        logInternal("error", format, $0)
    }
}
